spec/support/serializers.rb

```diff
@@ -92,6 +92,23 @@ def type
     end
   end
 
+  class SimplestPostSerializerWithoutSelfLink
```

Comment:
There is already a `PostSerializerWithoutLinks` and tests for self_link behaviors. So, you can drop this new class and just use `PostSerializerWithoutLinks` in the test instead.